tests/test_equine_protonet.py

```diff
@@ -225,46 +224,57 @@ def test_predict_fail_before_training(random_dataset):
 @given(random_dataset=random_dataset())
 @settings(deadline=None, max_examples=1)
 def test_equine_protonet_save_load(random_dataset) -> None:
-    dataset, num_classes, _ = random_dataset
-    X, Y = dataset.tensors
-    embedding_model = BasicEmbeddingModel(X.shape[1], num_classes)
+    dataset, num_classes, X, embedding_model = use_basic_embedding_model(random_dataset)
 
     model = eq.EquineProtonet(embedding_model, num_classes)
     model.train_model(dataset, num_episodes=10)
 
-    old_output = model.predict(X[1:10])
-    tmp_filename = "tmp_eq_proto.pt"
-    if os.path.exists(tmp_filename):
-        os.remove(tmp_filename)
-    model.save(tmp_filename)
-    new_model = eq.EquineProtonet.load(tmp_filename)
-    new_output = new_model.predict(X[1:10])
-    assert (
-        torch.nn.functional.mse_loss(old_output.classes, new_output.classes) <= 1e-7
-    ), "Predictions changed on reload"
+    new_model, tmp_filename = use_save_load_model_tests(model, X)
+
     if os.path.exists(tmp_filename):
         os.remove(tmp_filename)  # Cleanup
 
 
 @given(random_dataset=random_dataset())
 @settings(deadline=None, max_examples=1)
 def test_equine_protonet_save_load_with_temperature(random_dataset) -> None:
-    dataset, num_classes, _ = random_dataset
-    X, Y = dataset.tensors
-    embedding_model = BasicEmbeddingModel(X.shape[1], num_classes)
+    dataset, num_classes, X, embedding_model = use_basic_embedding_model(random_dataset)
 
     model = eq.EquineProtonet(embedding_model, num_classes, use_temperature=True)
     model.train_model(dataset, num_episodes=10)
 
-    old_output = model.predict(X[1:10])
-    tmp_filename = "tmp_eq_proto.pt"
+    new_model, tmp_filename = use_save_load_model_tests(model, X)
+
     if os.path.exists(tmp_filename):
-        os.remove(tmp_filename)
-    model.save(tmp_filename)
-    new_model = eq.EquineProtonet.load(tmp_filename)
-    new_output = new_model.predict(X[1:10])
-    assert (
-        torch.nn.functional.mse_loss(old_output.classes, new_output.classes) <= 1e-7
-    ), "Predictions changed on reload"
+        os.remove(tmp_filename)  # Cleanup
+
+
+@given(random_dataset=random_dataset())
+@settings(deadline=None, max_examples=1)
+def test_equine_protonet_save_load_with_feature_and_label_names(random_dataset) -> None:
+    dataset, num_classes, X, embedding_model = use_basic_embedding_model(random_dataset)
+
+    # without feature and label names
+    model = eq.EquineProtonet(embedding_model, num_classes)
```

Comment:
Change the number epochs to like 2 or something small since we don't care about training here